aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -915,15 +920,44 @@ std::tuple<Tensor, Tensor, Tensor> _lu_with_info_cuda(const Tensor& self, bool p
     self_working_copy = at::empty_like(self);
   } else {
     self_working_copy = cloneBatchedColumnMajor(self);
+    auto batch_size = batchCount(self);
     AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "lu_cuda", [&]{
       apply_lu<scalar_t>(self_working_copy, pivots_tensor, infos_tensor, pivot);
+      if (self.dim() > 2 && pivot && m == n && m <= 32) {
```

Comment:
I would rather we do this separately after this dispatch like so:
```c++
AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "lu_cuda", [&]{
  apply_lu<scalar_t>(self_working_copy, pivots_tensor, infos_tensor, pivot);
});

if (self.dim() > 2 && pivot && m == n && m <= 32) {
  if ((infos_tensor > 0).any()) {  // This checking is faster
    // Perform new dispatch here
  }
}
```

Do you think this is OK?

Reply:
Moving the new dispatch to a separate block requires wrapping the block with `AT_DISPATCH_FLOATING_TYPES` again. Unless this has some advantages, I would not do that to reduce the generated code.

The faster-checking suggestion may mean more work as not all matrices will be singular. Currently, only the singular matrices are recalculated.

Reply:
I do agree about the new dispatch generating new code unnecessarily - thank you for pointing that out.

Re the faster checking condition, yes, you are right there too. My bad.